### ISSUE TITLE
v1.19.0 DO convert fails with stack trace

### ISSUE THREAD

**Issue by DumpySquare** [f5devcentral/f5-automation-config-converter#85]
### Environment
 * ACC Version:  1.19.0
 * AS3 Version:  n/a
 * BIG-IP Version:  12.x?

### Summary
DO conversion results in stack error

### Steps To Reproduce

Below is my general test config found here:  https://github.com/DumpySquare/f5-fasting/blob/master/ACC/base_do.conf
1. Convert following config:
```conf
sys global-settings { 
    console-inactivity-timeout 1200
    file-whitelist-path-prefix "{/var/local/scf} {/tmp/} {/shared/} {/config/} {/usr/share/aws/} {/var/config/rest/downloads/appsvcs_update.cli}"
    gui-setup disabled
    hostname devCloud01.benlab.io
    mgmt-dhcp dhcpv6
 }
sys software update { 
    auto-check enabled
    auto-phonehome disabled
    frequency weekly
 }
sys management-ip 10.200.244.110/24 { 
    description configured-statically
 }
sys provision apm { 
    level nominal
 }
sys provision avr { 
    level nominal
 }
sys provision ltm { 
    level nominal
 }
sys ntp { 
    timezone US/Central
 }
sys dns { 
    description configured-by-dhcp
    name-servers { 192.168.200.7 192.168.200.8 }
    search { benlab.io }
 }
net dns-resolver /Common/f5-aws-dns { 
    forward-zones {
        amazonaws.com {
            nameservers {
                8.8.8.8:53 { }
            }
        }
        idservice.net {
            nameservers {
                8.8.8.8:53 { }
            }
        }
    }
    route-domain /Common/0
 }
net vlan /Common/internal { 
    interfaces {
        1.0 { }
    }
    tag 4094
 }
analytics global-settings {   }
sys management-route /Common/default { 
    description configured-statically
    gateway 10.200.244.1
    network default
 }
auth source { 
    fallback true
 }
auth remote-user { 
    default-role admin
    remote-console-access tmsh
 }
auth radius /Common/system-auth { 
    servers {
        /Common/system_auth_name1
    }
 }
auth radius-server /Common/system_auth_name1 { 
    secret $M$mn$9uYx+bTjLD8YSGZAUEfFwHvpSDwZsL25kZxdn5NZmCI=
    server 10.200.244.1
 }
net route-domain /Common/0 { 
    id 0
    vlans {
        /Common/http-tunnel
        /Common/socks-tunnel
        /Common/internal
    }
 }
sys snmp { 
    agent-addresses { tcp6:161 udp6:161 }
    communities {
        /Common/comm-public {
            community-name public
            source default
        }
    }
    disk-monitors {
        /Common/root {
            minspace 2000
            path /
        }
        /Common/var {
            minspace 10000
            path /var
        }
    }
    process-monitors {
        /Common/bigd {
            max-processes infinity
            process bigd
        }
        /Common/chmand {
            process chmand
        }
        /Common/httpd {
            max-processes infinity
            process httpd
        }
        /Common/mcpd {
            process mcpd
        }
        /Common/sod {
            process sod
        }
        /Common/tmm {
            max-processes infinity
            process tmm
        }
    }
 }
sys snmp { 
    agent-addresses { tcp6:161 udp6:161 }
    communities {
        /Common/comm-public {
            community-name public
            source default
        }
    }
    disk-monitors {
        /Common/root {
            minspace 2000
            path /
        }
        /Common/var {
            minspace 10000
            path /var
        }
    }
    process-monitors {
        /Common/bigd {
            max-processes infinity
            process bigd
        }
        /Common/chmand {
            process chmand
        }
        /Common/httpd {
            max-processes infinity
            process httpd
        }
        /Common/mcpd {
            process mcpd
        }
        /Common/sod {
            process sod
        }
        /Common/tmm {
            max-processes infinity
            process tmm
        }
    }
 }
sys httpd { 
    auth-pam-idle-timeout 12000
    ssl-port 8443
 }
net tunnels tunnel /Common/http-tunnel { 
    description "Tunnel for http-explicit profile"
    profile /Common/tcp-forward
 }
net tunnels tunnel /Common/socks-tunnel { 
    description "Tunnel for socks profile"
    profile /Common/tcp-forward
 }
security firewall port-list /Common/_sys_self_allow_tcp_defaults { 
    ports {
        22 { }
        53 { }
        161 { }
        443 { }
        1029-1043 { }
        4353 { }
    }
 }
security firewall port-list /Common/_sys_self_allow_udp_defaults { 
    ports {
        53 { }
        161 { }
        520 { }
        1026 { }
        4353 { }
    }
 }
```

2. Observe the following error message:
```out
5 [2022-03-12T13:20:09.512Z] [INFO]: f5 atc schema injected
6 [2022-03-12T13:25:44.523Z] [INFO]: f5.chariot.convertDO called
7 [2022-03-12T13:25:46.053Z] [INFO]: f5.chariot.convert text found
8 [2022-03-12T13:25:46.639Z] [ERROR]: f5.chariot.convertDO failed with [TypeError: key.split(...).at is not a function
	at /home/ted/projects/vscode-f5-chariot/node_modules/f5-automation-config-converter/src/engines/doConverter.js:290:54
	at Array.forEach (<anonymous>)
	at module.exports (/home/ted/projects/vscode-f5-chariot/node_modules/f5-automation-config-converter/src/engines/doConverter.js:112:26)
	at mainRunner (/home/ted/projects/vscode-f5-chariot/node_modules/f5-automation-config-converter/src/main.js:43:22)
	at Object.mainAPI (/home/ted/projects/vscode-f5-chariot/node_modules/f5-automation-config-converter/src/main.js:152:30)
	at /home/ted/projects/vscode-f5-chariot/out/extension.js:120:61
	at async /home/ted/projects/vscode-f5-chariot/out/extension.js:111:20]
```

### Expected Behavior
convert base tmos config to DO declaration without error

### Actual Behavior
See above error message.

**Comment by p-semenov-f5** [comment]
Did you try to convert without chariot?

I don't see that error during DO convertion:

2022-03-16 14:36:10 INFO 23 BIG-IP objects detected total
2022-03-16 14:36:11 INFO 11 DO stanzas generated

**Comment by DumpySquare** [comment]
yes, it seems the docker conversion works.  ACC 1.19 AS3 conversion through chariot also works.  Do we know what might be the difference?

This shows the output of converting the above base config with for AS3 (expected->empty declaration)
![image](https://user-images.githubusercontent.com/3260966/158800050-5543c23b-6763-4067-9205-628c870e94c1.png)

this shows the function call and input
![image](https://user-images.githubusercontent.com/3260966/158799874-89a211ca-66c5-4b37-afbd-e35ab718a67d.png)

**Comment by p-semenov-f5** [comment]
I don't know about Chariot. It is not our project.

**Comment by p-semenov-f5** [comment]
It is working for chariot 1.18:

1 [2022-03-17T17:12:07.773Z] [INFO]: ACC Details:  {
  name: 'f5-automation-config-converter',
  author: { name: 'F5 Networks' },
  description: 'F5 Automation Config Converter',
  version: '1.18.0',
  license: 'Apache-2.0',
  repository: 'git+https://github.com/f5devcentral/f5-automation-config-converter.git'
}
<...>
5 [2022-03-17T17:12:08.632Z] [INFO]: f5 atc schema injected
6 [2022-03-17T17:12:18.208Z] [INFO]: f5.chariot.convertDO called
7 [2022-03-17T17:12:18.235Z] [INFO]: f5.chariot.convert text found
8 [2022-03-17T17:12:18.261Z] [INFO]: ACC METADATA {
  recognized: undefined,
  declarationInfo: undefined,
  supported: undefined,
  unSupported: undefined
}

**Comment by p-semenov-f5** [comment]
Note for others:
We found that issue coming from different node versions. I used v16.13.1.

In old node version you could get element by index:  a[-1]
In new version you should use:                                    a.at(-1)

**Comment by p-semenov-f5** [comment]
fixed in 1.20
